web/source/text/processor.ts

```diff
@@ -241,7 +254,8 @@ namespace com.keyman.text {
 
       // Pass this key code and state to the keyboard program
       if(activeKeyboard && keyEvent.Lcode != 0) {
-        matchBehavior = kbdInterface.processKeystroke(fromOSK ? keyman.util.device : keyman.util.physicalDevice, outputTarget, keyEvent);
+        this.installInterface();
```

Comment:
Why is the interface installed at this point? Just so the keyboard can get at it? Should it be removed afterwards or always left available?

Remember that websites may also use `KeymanWeb` global...

Reply:
This is insurance against something I've seen in unit tests when things break a bit.

Currently, when a KMW `shutdown` doesn't go through properly or completely, sometimes we end up with parallel versions of KMW running, and an old, partially-shutdown one will "snipe" a command meant for the most-recent one's test.  So, `install`ing here ensures that the active `Processor` has its matching `KeyboardInterface` ready, even should that occur.

For general use cases (i.e., a single instance of KMW), this will simply repeatedly install the same `KeyboardInterface`, with nothing actually being changed.

----

As for
> Should it be removed afterwards or always left available?

I'd answer that with
> Remember that websites may also use KeymanWeb global...

This is why I decided against removing it after the `install` - there will always be a global copy to accept our more generalized (and documented) api, such as `registerStub` - and most importantly, `registerKeyboard`.

Reply:
Thanks -- can you add your explanation into the code as a comment? It'll help for the future!